### PR TITLE
SpanQualifier trainable component

### PR DESCRIPTION
**Pull request by percevalw**
## Description

This PR adds a new trainable `SpanQualifier` component with:
- multi nested label/qualifier and multi-class support (e.g. `_.negated`, `_.history`, `_.date.mode`, ...)
- qualifier to NER constraints, such as `_.date` only on `date` labelled spans

## Checklist

- [x] If this PR is a bug fix, the bug is documented in the test suite.
- [x] Changes were documented in the changelog (pending section).
- [x] If necessary, changes were made to the documentation (eg new pipeline).

### CODE REVIEW

**Comment by percevalw** [comment]
In particular @Thomzoy, I'm not quite confident about the parameters / API of the `make_candidate_getter` function here https://github.com/aphp/edsnlp/blob/b775d5a82b3de77f4025db009a49906980a3ae72/edsnlp/pipelines/trainable/span_qualifier/utils.py#L22-L32. You've worked on a similar issue here https://github.com/aphp/edsnlp/blob/a0b6021e575d9bd3ec10bda4e1a7c622fc1d8be4/edsnlp/pipelines/base.py#L30-L45
as well as in eCCI and with @aricohen93, what do you think ?
We also have multiple occurence of ~`on_ents`, ~`on_spans_groups` throughout the repository (bratconnector, dates, ...)

File: edsnlp/pipelines/trainable/span_qualifier/span_qualifier.py [hunk 1]
```diff
@@ -0,0 +1,532 @@
...
+        ]
+        # groups_indices:
+        #   num binding_groups (e.g. ["events", "negation"])
+        # * num label combinations in this group
+        # * presence or absence (bool) of the bindings of this groups in the combination
+        groups_bindings_indices = [
+            [self.bindings.index(binding) for binding in group_bindings]
+            for group_bindings in groups_bindings
+        ]
+        print("GROUPS INDICES", groups_bindings_indices)
```
Comment (Thomzoy) [inline edsnlp/pipelines/trainable/span_qualifier/span_qualifier.py:424]:
```suggestion
```

**Review by Thomzoy — COMMENTED** [review]
Very nice ! A few general remarks as discussed:
- It could be clearer to move `span_multiclassifier.py` (and in fact `stack_crf_ner.py`) from `edsnlp/models` into their corresponding `pipelines` directory, and move `pytorch_wrapper.py` directly into `edsnlp` ?
- It would be nice to have a tutorial/config with BERT embeddings for the nested NER and/or the span qualifier, as it would be a common use case


Also as discussed regarding the various `on_ents`, `on_span_groups` parameters scattered accross the codebase, let us do a uniformization PR after the main pending PRs are merged ?

**Comment by percevalw** [comment]
Thanks ! Indeed, a tutorial on training models would be nice 👍 I suggest we do that in another PR

**Pull request merged**